streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -1107,9 +1107,8 @@ boolean sendingOldValueEnabled() {
         );
 
         final StoreBuilder<TimestampedKeyValueStore<K, VR>> resultStore =
-            materializedInternal.queryableStoreName() == null
-                ? null
-                : new TimestampedKeyValueStoreMaterializer<>(materializedInternal).materialize();
+            new TimestampedKeyValueStoreMaterializer<>(materializedInternal).materialize();
```

Comment:
It might be easier to create the store within `TableProcessorNode#writeToTopology()` and encapsulate the decision if we need to materialize on not there?

Reply:
I took your other advice, which cleaned up this PR _a lot_. For this one, though, I think it's better this way. Note that `materialize` doesn't really "materialize" anything, it just converts the Materialized config object into a StoreBuilder topology object. It should be harmless to create a StoreBuilder that we never actually use.

This approach lets us keep Materialized in the DSL "parsing" layer, which is otherwise universally where it lives.

Reply:
SG